common/src/borrow.rs

```diff
@@ -203,14 +204,13 @@ impl<M: Borrow<Market>> LinkedBorrowPosition<M> {
         let duration_ms = Decimal::from(env::block_timestamp_ms() - last_snapshot.timestamp_ms.0);
         let ms_in_a_year = Decimal::from(MS_IN_A_YEAR);
         let interest_rate_part = interest_rate * duration_ms / ms_in_a_year;
-        let interest = interest_rate_part
-            * Decimal::from(self.position.get_borrow_asset_principal().as_u128());
+        let interest = interest_rate_part * self.position.get_borrow_asset_principal().to_decimal();
 
         interest.to_u128_ceil().unwrap().into()
     }
 
     pub(crate) fn calculate_interest(&self, limit: u32) -> AccumulationRecord<BorrowAsset> {
-        let principal = Decimal::from(self.position.get_borrow_asset_principal().as_u128());
+        let principal = self.position.get_borrow_asset_principal().to_decimal();
```

Comment:
If this had a `From<...> for Decimal` impl block it could be replaced with:
```suggestion
        let principal = self.position.get_borrow_asset_principal().into();
```
which might not be as explicit for some folk, but I find it more idiomatic.

Reply:
https://github.com/Templar-Protocol/contract-mvp/pull/59